### ISSUE TITLE
o-autocomplete: change label to an explicit label - ID: DAC_ Headings_And_Labels_01

### ISSUE THREAD

**Issue by JakeChampion** [Financial-Times/origami#497]
# Issue

This is from page 14 of the [audit report](https://drive.google.com/file/d/17wq6b0CIX41RA4muBFDzUDxE0-qzHBIH/view)

Issue can be found on: <https://www.ft.com/__origami/service/build/v3/demo?component=o-autocomplete%401.3.0&demo=static&system_code=origami-repo-data&brand=master>

The ‘Select your country’ input has been labelled implicitly. 
This issue may affect users of assistive technologies, for screen reader users the label is read as ‘Select your country Clear input edit combo’ and voice activation users are unable to use the visible label ‘Select your country’ to interact with the component.
The input field and all of its content has been wrapped inside the label tag and when users enter a letter into the input field a button appears in the field which is also labelled using the HTML title tag and a visually hidden label of ‘Clear input’.
When users attempt to activate the ‘Clear input Button’ using keyboard and mouse commands nothing happens and the entries in the form are not cleared.

# Voice Activation user comments:
>The box does not respond to label by name. I must use the command ‘Click box’ to move my focus into the field. Users would expect it to respond to ‘Select your country’ as it has a visual label.

# Proposed Solution:
Use an explicit label for the input field which has matching ‘for’ and ‘id’ attributes, this will enable voice activation users to use the visual label to enter the input. Place the ‘Clear input’ button adjacent to the input field with a visual label which will enable users of assistive technologies to identify and interact with the button correctly.

**Comment by JakeChampion** [comment]
This issue is also present on <https://www.ft.com/__origami/service/build/v3/demo?component=o- autocomplete%401.3.0&demo=dynamic&system_code=origami-repo-data&brand=master> and found on page 17 of the audit report

**Comment by akomiqaia** [comment]
We added explicit labels for all demo templates in [this PR](https://github.com/Financial-Times/origami/pull/811) but issue related to `Clear input` still remains. We opened a [new issue](https://github.com/Financial-Times/origami/issues/817) just for `Clear input` since this might need to get confirmed by design team.
